packages/cloud_firestore/cloud_firestore/example/test_driver/cloud_firestore_e2e.dart

```diff
@@ -1,12 +1,549 @@
+import 'dart:async';
+
 import 'package:e2e/e2e.dart';
 import 'package:flutter_test/flutter_test.dart';
+import 'package:firebase_core/firebase_core.dart';
 import 'package:cloud_firestore/cloud_firestore.dart';
 
 void main() {
   E2EWidgetsFlutterBinding.ensureInitialized();
 
-  testWidgets('Can initialize the plugin', (WidgetTester tester) async {
-    Firestore firestore = Firestore();
-    expect(firestore, isNotNull);
+  group('$Firestore', () {
+    Firestore firestore;
+    Firestore firestoreWithSettings;
+
+    setUpAll(() async {
+      final FirebaseOptions firebaseOptions = const FirebaseOptions(
+        googleAppID: '1:79601577497:ios:5f2bcc6ba8cecddd',
+        gcmSenderID: '79601577497',
+        apiKey: 'AIzaSyArgmRGfB5kiQT6CunAOmKRVKEsxKmy6YI-G72PVU',
+        projectID: 'flutter-firestore',
+      );
+      final FirebaseApp app = await FirebaseApp.configure(
+        name: 'test',
+        options: firebaseOptions,
+      );
+      final FirebaseApp app2 = await FirebaseApp.configure(
+        name: 'test2',
+        options: firebaseOptions,
+      );
+      firestore = Firestore(app: app);
+      firestoreWithSettings = Firestore(app: app2);
+      await firestoreWithSettings.settings(
+        persistenceEnabled: true,
+        host: null,
+        sslEnabled: true,
+        cacheSizeBytes: 1048576,
+      );
+    });
+
+    test('getDocumentsWithFirestoreSettings', () async {
+      final Query query = firestoreWithSettings.collection('messages').limit(1);
+      final QuerySnapshot querySnapshot = await query.getDocuments();
+      expect(querySnapshot.documents.length, 1);
+    });
+
+    test('getDocumentsFromCollection', () async {
+      final Query query = firestore
+          .collection('messages')
+          .where('message', isEqualTo: 'Hello world!')
+          .limit(1);
+      final QuerySnapshot querySnapshot = await query.getDocuments();
+      expect(querySnapshot.metadata, isNotNull);
+      expect(querySnapshot.documents.first['message'], 'Hello world!');
+      final DocumentReference firstDoc =
+          querySnapshot.documents.first.reference;
+      final DocumentSnapshot documentSnapshot = await firstDoc.get();
+      expect(documentSnapshot.data['message'], 'Hello world!');
```

Comment:
subjective nit: empty line between `action/assert` blocks for better readability.

Reply:
the ones below too. e.g.

```dart
final DocumentSnapshot cachedSnapshot = await firstDoc.get(source: Source.cache);
expect(cachedSnapshot.data['message'], 'Hello world!');

final DocumentSnapshot snapshot = await firstDoc.snapshots().first;
expect(snapshot.data['message'], 'Hello world!');
```